common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java

```diff
@@ -94,6 +96,23 @@ public boolean preferDirectBufs() {
     return conf.getBoolean(SPARK_NETWORK_IO_PREFERDIRECTBUFS_KEY, true);
   }
 
+  /** The threshold for consolidation, it is derived upon the memoryOverhead in yarn mode. */
```

Comment:
This replicates a lot of logic from elsewhere with hard-coded constants. Is it really important vary it so finely and add a whole new conf? It seems like this ought to be pretty independent of the environment, whether consolidating a buffer of size X is worthwhile.

Reply:
@srowen Thanks, yes, I think you are right, we can just make it some fixed factor of the frame size.